webui/src/pages/user/user-namespace-extension-list-item.tsx

```diff
@@ -0,0 +1,76 @@
+/********************************************************************************
+ * Copyright (c) 2020 TypeFox and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ********************************************************************************/
+
+import * as React from 'react';
+import { Extension } from '../../extension-registry-types';
+import { Paper, Typography, Theme, createStyles, WithStyles, withStyles } from '@material-ui/core';
+import { createRoute } from '../../utils';
+import { ExtensionDetailRoutes } from '../extension-detail/extension-detail';
+import { Link as RouteLink } from 'react-router-dom';
+
+const itemStyles = (theme: Theme) => createStyles({
+    link: {
+        textDecoration: 'none',
+    },
+    paper: {
+        display: 'flex',
+        flexDirection: 'column',
+        textAlign: 'center',
+        padding: theme.spacing(2, 3)
+    },
+    extensionLogo: {
+        display: 'block',
+        marginBottom: '1rem',
+        height: '5rem'
+    },
+    p: {
+        display: 'flex',
+        justifyContent: 'space-between',
+    }
+});
+
+const UserNamespaceExtensionListItemComponent = ({ extension, classes }: UserNamespaceExtensionListItemComponent.Props) => {
+    const route = extension && createRoute([ExtensionDetailRoutes.ROOT, extension.namespace, extension.name]) || '';
+    return (
+        extension ? (
+            <RouteLink to={route} className={classes.link}>
+                <Paper className={classes.paper}>
+                    <img
+                        src={extension.files.icon}
+                        alt={extension.displayName || extension.name}
+                        className={classes.extensionLogo}
+                    />
+                    <div>
+                        <Typography variant='h6' noWrap style={{ fontSize: '1.15rem' }}>
+                            {extension.displayName || extension.name}
+                        </Typography>
+                        <p className={classes.p}>
+                            <span>Latest Version:</span>
+                            <span>{extension.version}</span>
+                        </p>
+                        <p className={classes.p}>
+                            <span>Released on:</span>
+                            <span>{(new Date(extension.timestamp || '')).toLocaleString()}</span>
+                        </p>
+                    </div>
+                </Paper>
```

Comment:
Could we make these cards more compact? There's a lot of space in them. Maybe put the icon left and name / info right?

Reply:
Thanks. I have made the change this is how they look now:

![image](https://user-images.githubusercontent.com/46004116/81879201-88781380-95a3-11ea-8d90-60082222e591.png)

Reply:
Two more problems:
 * The default icon defined in PageSettings should be used if an extension does not have an icon.
 * The extension name shall be shortened with an ellipsis if necessary

<img width="756" alt="Screen Shot 2020-05-14 at 15 25 41" src="https://user-images.githubusercontent.com/4067210/81940164-96f01a80-95f7-11ea-8fca-d72bf3bbf3ee.png">